src/Crate/PDO/Http/Client.php

```diff
@@ -120,6 +175,54 @@ public function setHttpBasicAuth($username, $passwd)
      */
     public function setHttpHeader($name, $value)
     {
-        $this->client->setDefaultOption('headers/'.$name, $value);
+        foreach ($this->serverPool as $k => &$s) {
+            /**
+             * @var $s Server
+             */
+            $s->setHttpHeader($name, $value);
+        }
     }
+
+    /**
+     * @return Server The next available server instance
+     */
+    private function nextServer()
+    {
+        $server = $this->availableServers[0];
+        $this->roundRobin();
+        return $server;
+    }
+
+    /**
+     * Very simple round-robin implementation
+     * Pops the first item of the availableServers array and appends it at the end.
+     *
+     * @return void
+     */
+    private function roundRobin()
+    {
+        /**
+         * Performing round robin on the array only makes sense
+         * if there are more than 1 available servers.
+         */
+        if (count($this->availableServers) > 1) {
+            $this->availableServers[] = array_shift($this->availableServers);
+        }
+    }
+
+    /**
+     * @param string           $server
+     * @param ConnectException $exception
+     */
+    private function dropServer($server, $exception)
+    {
+        if (($idx = array_search($server, $this->availableServers)) !== false) {
+            unset($this->availableServers[$idx]);
+        }
+        if (count($this->availableServers) == 0) {
```

Comment:
as already discussed: this code will break the loop in sqlRequest, which I didn't see first and we agreed that this is not the best design ;) please move it...